src/sparseml/transformers/sparsification/trainer.py

```diff
@@ -758,8 +736,10 @@ def check_disable(self, epoch: float, force: bool = False):
             self.disable_amp(epoch)
 
     def qat_active(self, epoch: float) -> bool:
-        return (self.trainer.manager and self.trainer.manager.qat_active(epoch)) or any(
-            bool(man.quantization_modifiers) for man in self.trainer.arch_managers
+        return (
+            self.trainer.manager
+            and self.trainer.manager.qat_active(epoch)
+            or self.trainer.arch_manager.quantization_modifiers
```

Comment:
There may be a potential error case here. If `self.trainer.manager.qat_active(epoch)` is false and `arch_manager` is None, then trying to access self.trainer.arch_manager.quantization_modifiers will throw an error. Is such a case possible?

Reply:
good corner-case hunting! testing